src/scene/light.js

```diff
@@ -75,6 +75,19 @@ class LightRenderData {
 
         // visible shadow casters
         this.visibleCasters = [];
+
+        // an array of view bind groups, single entry is used for shadows
+        /** @type {import('../platform/graphics/bind-group.js').BindGroup[]} */
+        this.viewBindGroups = [];
+    }
+
+    // releases GPU resources
+    destroy() {
+        this.viewBindGroups.forEach((bg) => {
+            bg.defaultUniformBuffer.destroy();
+            bg.destroy();
+        });
+        this.viewBindGroups.length = 0;
```

Comment:
Wouldn't it make sense to delete `bg.defaultUniformBuffer` in `BindGroup#destroy` instead? It feels like `BindGroup#destroy` isn't doing a full job when you still need to delete internal properties yourself first.

Reply:
It's a property value I assign to it, but it does not own it, as multiple bindGroups can use the same uniform buffer. If JS had some easier way to do ref counting, this would be ref counted perhaps.
It's similar to how when you create a render target, you give it color buffer, but you need to destroy it even when RT itself is destroyed.

Reply:
actually, a material -> texture is even more similar to this. Texture, as well as Uniform Buffer is a buffer GPU resource. Material does not own it, it does not destroy it, only uses it. Same here. In this specific instance, there is only one user of this uniform buffer, so I destroy it, but that's not a case in general.